app/main.py

```diff
@@ -1 +1,42 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self._hidden: bool = False
+        Animal.alive.append(self)
+
+    @property
+    def hidden(self) -> bool:
+        return self._hidden
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
```

Comment:
Code Efficiency: Use the class name to get access to the class attributes. You should use `Herbivore` instead of `self` in the `hide` method.